server/src/utils/fight/fightMethods.ts

```diff
@@ -1430,13 +1495,22 @@ const block = ({
 
   let opponentBlock = getFighterStat(opponent, 'block');
 
-  // increase block if blocking a throwing a weapon with `Hideaway`
+  // increase block if blocking a throwing weapon with `Hideaway`
+  // multiply by ease so the real block bonus mirrors the displayed bonus
   if (thrown && opponent.skills.find((sk) => sk.name === SkillName.hideaway)) {
-    opponentBlock += SkillModifiers[SkillName.hideaway][FightStat.BLOCK]?.percent ?? 0;
+    opponentBlock += ease * (SkillModifiers[SkillName.hideaway][FightStat.BLOCK]?.percent ?? 0);
+  }
+
+  // increase block if 1HP and `Survival`
+  if (opponent.hp === 1 && opponent.skills.find((sk) => sk.name === SkillName.survival)) {
+    opponentBlock += SkillModifiers[SkillName.survival][FightStat.BLOCK]?.percent ?? 0;
   }
 
   return Math.random() * ease
-    < (opponentBlock - getFighterStat(fighter, 'accuracy'));
+    < Math.min(
+      opponentBlock - getFighterStat(fighter, 'accuracy'),
+      0.9 * ease,
```

Comment:
Why add a max of 0.9 * ease here?

Reply:
Because with the survival update, block can effectively go to more than 100%, which can cause never ending fights. So I adopted the same solution as for evasion.